From 2afdc6934fabc2f3bd51052f7a9c549b1aa0e900 Mon Sep 17 00:00:00 2001
From: "vasudevan,krishna prasath" <krishna.prasathx.k.vasudevan@intel.com>
Date: Thu, 23 Jul 2015 00:38:56 +0700
Subject: [PATCH 349/389] USB:The fix for Host re-enumeration failure after
 sleep has been added (RTC 112108)

Host re-enumeration fails for few USB2.0 Mass storage devices after sleep,this is due
to the fact that for some mass storage devices time required to complete link training,
after VBUS is powered seems to be high,a similar kind of issue is faced for USB 3.0 and
fix is provided by linux maintainers.The fix is restricted to USB 3.0 devices by adding
a check before timeout.The check for USB 3.0 devices is removed such that the driver will
allow enough time for the device to complete its link training for all USB devices.

Signed-off-by: vasudevan,krishna prasath <krishna.prasathx.k.vasudevan@intel.com>
Change-Id: If74ed02a9983a78725aeb3d49bb7d44d51f17c6f
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/5831
Reviewed-by: Fung, Joshua <joshua.fung@intel.com>
Reviewed-by: Clarke, Kevin1 <kevin1.clarke@intel.com>
---
 drivers/usb/core/hub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index d2bd9d7..b2b709d 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3320,7 +3320,7 @@ int usb_port_resume(struct usb_device *udev, pm_message_t msg)
 
 	clear_bit(port1, hub->busy_bits);
 
-	if (udev->persist_enabled && hub_is_superspeed(hub->hdev))
+	if (udev->persist_enabled)
 		status = wait_for_ss_port_enable(udev, hub, &port1, &portchange,
 				&portstatus);
 
-- 
1.9.3

